src/locktower.rs

```diff
@@ -0,0 +1,172 @@
+use std::collections::VecDeque;
+pub struct VoteLock {
+    /// Fork id, same as slot id. There can only be 1 fork per slot, since the empty PoH ticks
+    /// cannot be voted on.  A validator cannot vote on a fork unless that fork is connected to
+    /// a trunk. The starting trunk is the initial `finalized_fork`.
+    fork: u64,
+    /// Lockout is in terms of slots, so the slot height or fork id to unroll this
+    /// vote must be greater than fork + lockout.
+    lockout: u64,
+}
+
+pub enum LockTowerError {
+    VoteLocked,
+    VoteNotConverging,
+}
+
+pub struct LockTower {
+    vote_locks: VecDeque<VoteLock>,
+    /// The maximum size of vote_locks before reward is generated.  When the queue reached max_size,
+    /// a reward is generated and the oldest lock is removed.  That fork is considered finalized.
+    max_size: usize,
+    /// Depth is equal to lockout, since lockout doubles as votes get stacked
+    /// If the requirement is at depth 8 to have converge_amount of 0.667,
+    /// than the lockout will not increase past 256 (2^8) on the fork at depth 8
+    /// without 66.7% total network commitment to that fork.
+    ///
+    /// It is not important for the rest of the network to also have the same lockout on the fork.
+    /// The worst case is that the lockout for this node will double to 512 due to
+    /// rollback for the rest of the network.  So any lockout of network commitment is acceptable,
+    /// but it must cover the required threshold percentage of network stake.
+    converge_depth: usize,
```

Comment:
`depth` -> `height`?

Reply:
@garious depth in the locktower, not height in the ledger